padertorch/summary/tbx_utils.py

```diff
@@ -210,14 +263,56 @@ def audio(signal, sampling_rate: int = 16000, batch_first=False,
     return signal, sampling_rate
 
 
+def matplotlib_figure_to_image(
+        figure: 'matplotlib.figure.Figure' = None,
+) -> np.ndarray:
+    """
+    Converts a matplotlib figure to a numpy array that can be handled by
+    tensorboardX.
+
+    Note:
+        Matplotlib is in general quite slow, so you shouldn't be computing
+        matplotlib plots in every iteration. Use the `pt.Model.create_snapshot`
+        flag to reduce the computational overhead.
+
+    Warnings:
+        This function doesn't close the plot! Make sure that you close unused
+        plots to reduce memory consumption!
+
+    Args:
+        figure: matplotlib figure object. If `None`, it defaults to `plt.gcf()`.
+
+    Returns:
+        Numpy array with shape (color (4), width, height)
+
+    Examples:
+        >>> from matplotlib import pyplot as plt
+        >>> plt.plot([1, 3, 2, 4])  # doctest: +ELLIPSIS
+        [<matplotlib.lines.Line2D object ...>]
+        >>> image = matplotlib_figure_to_image()
+        >>> image.shape
+        (3, 480, 640)
+    """
+    from einops import rearrange
+
+    if figure is None:
+        from matplotlib import pyplot as plt
+        figure = plt.gcf()
+
+    figure.canvas.draw()
+    data = np.fromstring(figure.canvas.tostring_rgb(), dtype=np.uint8, sep='')
+    data = data.reshape(figure.canvas.get_width_height()[::-1] + (3,))
+    return rearrange(data, 't f c -> c t f')
+
+
 def review_dict(
         *,
-        loss: torch.Tensor=None,
-        losses: dict=None,
-        scalars: dict=None,
-        histograms: dict=None,
-        audios: dict=None,
-        images: dict=None,
+        loss: torch.Tensor = None,
+        losses: dict = None,
+        scalars: dict = None,
+        histograms: dict = None,
+        audios: dict = None,
+        images: dict = None,
```

Comment:
add `figures`?

Reply:
What's the difference to `images`?

Reply:
It supports matplotlib figures

Reply:
Hmm, now that I found this option, I'm not sure if the new function is required anymore. I don't see a strong reason to draw the figure in the review function rather than drawing it in the hook. If the user wants to draw the figure in review, the user can still use the `figure_to_image` function from tensorboardX.

It would be nice to document this somewhere.

Reply:
Added `figures`